static/main.ts

```diff
@@ -461,48 +478,42 @@ function removeOrphanedMaximisedItemFromConfig(config) {
 
     impl(config);
 
-    if (!found) {
+    if (!(found as boolean)) {
         config.maximisedItemId = null;
     }
 }
 
-function setupLanguageLogos(languages) {
-    _.each(
-        languages,
-        function (lang) {
-            try {
-                if (lang.logoUrl !== null) {
-                    lang.logoData = logos('./' + lang.logoUrl);
-                    if (lang.logoUrlDark !== null) {
-                        lang.logoDataDark = logos('./' + lang.logoUrlDark);
-                    }
+function setupLanguageLogos(languages: Record<LanguageKey, Language>) {
+    for (const lang of Object.values(languages)) {
+        try {
+            if (lang.logoUrl !== null) {
+                lang.logoData = logos('./' + lang.logoUrl);
+                if (lang.logoUrlDark !== null) {
+                    lang.logoDataDark = logos('./' + lang.logoUrlDark);
                 }
-            } catch (ignored) {
-                lang.logoData = '';
             }
-        },
-        this
-    );
+        } catch (ignored) {
+            lang.logoData = '';
+        }
+    }
 }
 
 function earlyGetDefaultLangSetting() {
-    // returns string | undefined
     return Settings.getStoredSettings().defaultLanguage;
 }
 
-function getDefaultLangId(subLangId, options) {
-    var defaultLangId = subLangId;
+function getDefaultLangId(subLangId: LanguageKey | undefined, options: CompilerExplorerOptions) {
+    let defaultLangId = subLangId;
     if (!defaultLangId) {
-        var defaultLangSetting = earlyGetDefaultLangSetting();
-        if (defaultLangSetting && options.languages[defaultLangSetting] !== undefined) {
+        const defaultLangSetting = earlyGetDefaultLangSetting();
+        if (defaultLangSetting && defaultLangSetting in options.languages) {
             defaultLangId = defaultLangSetting;
-        } else if (options.languages['c++']) {
+        } else if ('c++' in options.languages) {
             defaultLangId = 'c++';
         } else {
-            defaultLangId = _.keys(options.languages)[0];
+            defaultLangId = utils.keys(options.languages as unknown as Record<LanguageKey, Language>)[0];
```

Comment:
More shenanigans here, ts thinks `options.languages` is `never`. I'll play around with the conditions / typing of `options.langauges`